tests/interfaces/test_autograd.py

```diff
@@ -284,11 +285,9 @@ def cost(a, cache):
         # Without caching, 9 evaluations would be required to compute
         # the Jacobian: 1 (forward pass) + 2 (backward pass) * (2 shifts * 2 params)
         #
-        # However, the jacobian is being cached in the interface by default,
-        # hence we do 5 evaluations
         params = np.array([0.1, 0.2])
         qml.jacobian(cost)(params, cache=None)
-        assert dev.num_executions == 5
+        assert dev.num_executions == 9
```

Comment:
Is this a performance degradation?

Reply:
This matches the numbers for all the other interfaces for this test.  autograd had this extra level of caching that was always performed. Instead of using `batch_vjp`, it computed the full jacobian and then locally cached it. 

So kinda a performance degradation for autograd, but only when caching is off, and it now matches the behavior of the other interfaces.

Reply:
Just to add: 

- JAX, Torch, and TF all vectorize Jacobian computations, so they will have 5 executions without the cache.
- Autograd has a for loop over the cotangent computation, so has very poor performance compared to the others. It will actually _recompute_ the Jacobian within the loop, unfortunately.

Previously I added a hotfix to the interfaces specifically to Autograd, to avoid this behaviour, because it was noticeable to users ('autograd is making too many quantum executions').

I will tag @trbromley here

Reply:
So this _could_ be viewed as a performance degredation, depending on how we view the `cache=False` case.

Reply:
> So kinda a performance degradation for autograd, but only when caching is off, and it now matches the behavior of the other interfaces.

Just to add here: after discussing with @albi3ro, this isn't the case. Autograd uses 9 executions without caching, but the others still use 5 without caching.

Reply:
Let's discuss this but my preference is to maintain existing functionality as much as possible unless we have a good motivation to change it. So in this case, `assert dev.num_executions == 5` would be my preference.

Reply:
I'll try to figure out some workarounds.

Maybe in the meantime, we prioritize the torch interface instead?

Reply:
@trbromley @josh146 @dwierichs 

Caching worked out. Kinda hacky, but works.